tap_github/streams.py

```diff
@@ -906,6 +919,32 @@ def http_headers(self) -> dict:
     ).to_dict()
 
 
+class PullRequestFilesStream(GitHubStream):
+    """Defines 'PullRequestFiles' stream."""
+
+    name = "pull_request_files"
+    path = "/repos/{org}/{repo}/pulls/{pull_number}/files"
+    primary_keys = ["filename"]
+    parent_stream_type = PullRequestsStream
+    ignore_parent_replication_key = False
+    state_partitioning_keys = ["repo", "org"]
```

Comment:
I think you want something like this, but not 100% sure how you get the `pull_number`, you might need to add it to the record in `post_process`.
```suggestion
    state_partitioning_keys = ["repo", "org", "pull_number"]
```